py_gtfs_rt_ingestion/ingest.py

```diff
@@ -106,43 +108,61 @@ def main(files: list[str]) -> None:
-
-        with ThreadPoolExecutor(max_workers=POOL_SIZE) as executor:
-            for result in executor.map(
-                lambda x: gz_to_pyarrow(*x), [(f, config) for f in files]
-            ):
-                if isinstance(result, pa.Table):
-                    pa_table = pa.concat_tables([pa_table, result])
-                else:
-                    failed_ingestion.append(result)
-
-        logging.info(
-            "Completed converting %d files with config %s",
-            len(files),
-            config.config_type,
-        )
+    # filesystem to use when writing parquet files
+    s3_filesystem = fs.S3FileSystem()
 
-        logging.info("Writing Table to %s", export_bucket)
-        s3_filesystem = fs.S3FileSystem()
-        pq.write_to_dataset(
-            table=pa_table,
-            root_path=os.path.join(export_bucket, str(config.config_type)),
-            filesystem=s3_filesystem,
-            partition_cols=["year", "month", "day", "hour"],
-        )
+    try:
+        config_type = ConfigType.from_filename(files[0])
+        if config_type == ConfigType.SCHEDULE:
+            if len(files) != 1:
+                raise ArgumentException(
+                    "Received more than one schedule GTFS zip file"
+                )
+            logging.info("Reading %s file and converting to parquet", files[0])
+            for prefix, table in zip_to_pyarrow(files[0]):
+                pq.write_to_dataset(
+                    table=table,
+                    root_path=os.path.join(export_bucket, prefix),
+                    filesystem=s3_filesystem,
+                    partition_cols=["year", "month", "day"],
+                )
+        else:
+            config = Configuration(config_type)
+            pa_table = pa.table(
+                config.empty_table(), schema=config.export_schema
+            )
+
+            logging.info(
+                "Creating pool with %d threads, %d cores available",
+                POOL_SIZE,
+                os.cpu_count(),
+            )
+
+            with ThreadPoolExecutor(max_workers=POOL_SIZE) as executor:
+                for result in executor.map(
+                    lambda x: gz_to_pyarrow(*x), [(f, config) for f in files]
+                ):
+                    if isinstance(result, pa.Table):
+                        pa_table = pa.concat_tables([pa_table, result])
+                    else:
+                        failed_ingestion.append(result)
+
+            logging.info(
+                "Completed converting %d files with config %s",
+                len(files),
+                config.config_type,
+            )
+
+            logging.info("Writing Table to %s", export_bucket)
+            pq.write_to_dataset(
+                table=pa_table,
+                root_path=os.path.join(export_bucket, str(config.config_type)),
+                filesystem=s3_filesystem,
+                partition_cols=["year", "month", "day", "hour"],
+            )
```

Comment:
two track conversion here now.  one track for gtfs and one for gtfs_rt.

if we ever have more sources, we probably want to abstract this a bit and move the logic into the library.

tbh, i kind of feel like we should do that now.